### PR TITLE
feat: exposing ssl ca cert path to httpfs

### PR DESCRIPTION
**Pull request by pvaezi**
In a corporate environment where trying to connect to S3 bucket behind proxy and with custom certificate, I'm seeing the following error:

```
RuntimeError: IO Error: SSLConnection error for HTTP HEAD to 'https://....s3.amazonaws.com/...'
```

In order to access AWS we need to set our own custom certificate file. This PR is aiming to expose such parameters to the users of duckdb all the via to underlaying `cpp-httplib`.

My C++ is a bit rusty going back to it after many years, please guide me if I'm missing anything, but this is aiming solve the problem mentioned by many corporate users here: https://github.com/duckdb/duckdb/issues/6064

**How to test:**

```bash
GEN=ninja make BUILD_HTTPFS=1 -j8
./build/release/duckdb
```

```sql
D SET ca_cert_file='<cert.crt>';
D SET enable_server_cert_verification=True;
D LOAD './build/release/extension/httpfs/httpfs.duckdb_extension';
```

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks for the PR! Could you perhaps expand on what behavior this is enabling, and why this is desirable?

File: extension/httpfs/httpfs.cpp [hunk 1]
```diff
@@ -36,6 +36,8 @@ HTTPParams HTTPParams::ReadFrom(FileOpener *opener) {
 	float retry_backoff = DEFAULT_RETRY_BACKOFF;
 	bool force_download = DEFAULT_FORCE_DOWNLOAD;
 	bool keep_alive = DEFAULT_KEEP_ALIVE;
+	bool enable_server_cert_verification = DEFAULT_ENABLE_SERVER_CERT_VERIFICATION;
+	std::string_view ca_cert_file = DEFAULT_CA_CERT_FILE;
```
Comment (Tishj) [inline extension/httpfs/httpfs.cpp:40]: I dont think this should be a string_view, string is fine

Comment (pvaezi) [reply extension/httpfs/httpfs.cpp]: Thanks for the feedback. Fixed.

**Review by samansmink — APPROVED** [review]
LGTM!

**Comment by pvaezi** [comment]
> LGTM!

@samansmink Thanks for the approval, the pipeline is now passing, please let me know if any other updates needed from my end.

**Comment by szarnyasg** [comment]
I tagged this with the "Needs Documentation" label – note that at this moment, documentation is _not_ required for getting the code merged, this label is only used to indicate the need for a documentation page later.

**Comment by Mytherin** [comment]
Thanks for the PR! Could you resolve the merge conflict? This is likely conflicting with https://github.com/duckdb/duckdb/pull/10597

**Comment by pvaezi** [comment]
> Thanks for the PR! Could you resolve the merge conflict? This is likely conflicting with #10597

Thanks for the review, I rebased the branch. 

Also if there is instruction for updating documentation happy to take a stab at it in this PR or a separate one. Just need guidance to where the documentation resides.

**Comment by samansmink** [comment]
@pvaezi that would be much appreciated of course! the documentation lives here https://github.com/duckdb/duckdb-web feel free to open a PR there. There are contributing and building instructions in READMEs to help you out but things are pretty straightforward

**Comment by pvaezi** [comment]
> @pvaezi that would be much appreciated of course! the documentation lives here https://github.com/duckdb/duckdb-web feel free to open a PR there. There are contributing and building instructions in READMEs to help you out but things are pretty straightforward

Thanks for the pointer, made this PR: https://github.com/duckdb/duckdb-web/pull/2369

**Comment by pvaezi** [comment]
> @pvaezi that would be much appreciated of course! the documentation lives here https://github.com/duckdb/duckdb-web feel free to open a PR there. There are contributing and building instructions in READMEs to help you out but things are pretty straightforward

Based on PR https://github.com/duckdb/duckdb-web/pull/2369 feedback, the inclination was these parameters will automatically render as part of document parsing script, and the description seemed sufficient, if there are any other actions needed to get this PR merged please let me know.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**